mozetl/taar/taar_similarity.py

```diff
@@ -56,6 +58,48 @@ def get_samples(spark):
         )
     )
 
+    def is_user_addon(guid, addon):
+        # Note that the whitelist check on the last line has been removed
+        return not (
+            addon.is_system or
+            addon.app_disabled or
+            addon.type != "extension" or
+            addon.user_disabled or
+            addon.foreign_install
+        )
+
+    def is_whitelisted(guid, addon):
+        return not (
+            addon.is_system or
+            addon.app_disabled or
+            addon.type != "extension" or
+            addon.user_disabled or
+            addon.foreign_install or
+            guid not in whitelist
+        )
+
+    # vng: This adds 3 columns to maniuplate the sample to get just
+    # the client we are interested in.
+    # - active_user_addons: the list of all user addons
+    #   (non-system addons) that the user has installed
+    # - featured_addons: the list of featured addons installed
+    # - whitelisted: addons that are whitelisted
+    #
+    # Note that the filter here is what finally computes which clients
+    # meet our criteria
+    return samples_df.rdd
+            .withColumn('active_user_addons',
+                lambda p: [unicode(guid) for (guid, data) in p['active_addons'].items() if is_user_addon(guid, data)])
```

Comment:
@mlopatka I explicitly cast to unicode strings here mostly because I had surprising results when running set() operations.  I'm adding extra columns to the dataframe so that I can properly determine if i have at least one featured or whitelisted addon, or only whitelisted or only featured addons.